src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,124 @@
+package duke.parser;
+
+import duke.command.Command;
+import duke.exceptions.DukeException;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Deals with making sense of the user's commands.
+ *
+ * @author ish1506
+ */
+public class Parser {
+    private final String input;
+    private Command command;
+    private int taskIndex = -1;
+    private String[] args;
+
+    /**
+     * Constructs the <code>Parser</code> object given an input string.
+     *
+     * @param input the user input.
+     */
+    public Parser(String input) {
+        this.input = input;
+    }
+
+    public Command getCommand() {
+        return command;
+    }
+
+    public int getTaskIndex() {
+        return taskIndex;
+    }
+
+    public String[] getArgs() {
+        return args;
+    }
+
+    /**
+     * Parses the user input to get the command and its arguments.
+     *
+     * @throws DukeException if the input cannot be parsed.
+     */
+    public void parse() throws DukeException {
+        List<String> listOfInputs = Arrays.asList(input.split(" "));
+        if (input.isEmpty() || listOfInputs.isEmpty()) {
```

Comment:
Don't you think a series of switch case statements will be better than such large number of if else statements?